auditbeat/module/file_integrity/eventreader_ebpf.go

```diff
@@ -0,0 +1,123 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package file_integrity
+
+import (
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/elastic/beats/v7/auditbeat/internal/ebpf"
+	"github.com/elastic/ebpfevents"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+const clientName = "fim"
+
+type ebpfReader struct {
+	watcher ebpf.Watcher
+	done    <-chan struct{}
+	config  Config
+	log     *logp.Logger
+	eventC  chan Event
+	parsers []FileParser
+	paths   map[string]struct{}
+
+	_events <-chan ebpfevents.Event
+	_errors <-chan error
+}
+
+func (r *ebpfReader) Start(done <-chan struct{}) (<-chan Event, error) {
+	watcher, err := ebpf.GetWatcher()
+	if err != nil {
+		return nil, err
+	}
+	r.watcher = watcher
+	r.done = done
+
+	mask := ebpf.EventMask(ebpfevents.EventTypeFileCreate | ebpfevents.EventTypeFileRename | ebpfevents.EventTypeFileDelete)
+	r._events, r._errors = r.watcher.Subscribe(clientName, mask)
+
+	go r.consumeEvents()
+
+	r.log.Infow("started ebpf watcher", "file_path", r.config.Paths, "recursive", r.config.Recursive)
+	return r.eventC, nil
+}
+
+func (r *ebpfReader) consumeEvents() {
+	defer close(r.eventC)
+	defer r.watcher.Unsubscribe(clientName)
+
+	for {
+		select {
+		case event := <-r._events:
+			if event.Type != ebpfevents.EventTypeFileCreate &&
```

Comment:
maybe a switch case with ```default:``` logging the warning and continue improves readability?!